### PR TITLE
Use nosto-js for Nosto types

### PR DESCRIPTION
**Pull request by timowestnosto**
## Description

Get rid of client script types by using the types from nosto-js

## Checklist

- [ ] Pull Request is properly described.
- [ ] The corresponding Jira ticket is updated.
- [ ] I have requested a review from at least one reviewer.
- [ ] The changes are tested
- [ ] I have verified my UX changes with a designer
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -204,21 +158,6 @@
             "text"
         ]
     },
-    "codegen": {
```
Comment (timowestnosto) [inline package.json:207]: not needed anymore, since the search types are bundled in nosto-js

File: src/config.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ export function getDefaultConfig<State>() {
                     },
                     {
                         redirect: true,
-                        track: config.nostoAnalytics ? "serp" : undefined,
+                        track: (config.nostoAnalytics ? "serp" : undefined)!,
```
Comment (manikandan-ravikumar) [inline src/config.ts:105]: This signature looks a bit weird. The condition returns undefined but we are using bang operator on top of it

File: src/utils/state.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export const getStateActions = <State>({
                     ...config.fetch,
                 },
                 {
-                    track: config.nostoAnalytics ? "autocomplete" : undefined,
+                    track: (config.nostoAnalytics ? "autocomplete" : undefined)!,
```
Comment (manikandan-ravikumar) [inline src/utils/state.ts:60]: This signature looks a bit weird. The condition returns undefined but we are using bang operator on top of it

File: src/search.ts [hunk 4]
```diff
@@ -101,7 +101,7 @@ export async function search(
                 from,
             },
         },
-        { redirect, track, isKeyword }
+        { redirect, track: track!, isKeyword }
```
Comment (manikandan-ravikumar) [inline src/search.ts:104]: Should we instead handle invalid `track` values?

**Review by manikandan-ravikumar — APPROVED** [review]

**Pull request merged**